devops/scripts/benchmarks/benches/gromacs.py

```diff
@@ -0,0 +1,286 @@
+
+        # Generate configuration files
+        self.conf_result = run(
+            cmd_list,
+            add_sycl=True,
+        )
+
+    def run(self, env_vars):
+        if not self.gmx_path.exists():
+            raise FileNotFoundError(f"gmx executable not found at {self.gmx_path}")
+
+        model_dir = self.grappa_dir / self.model
+
+        if not model_dir.exists():
+            raise FileNotFoundError(f"Model directory not found: {model_dir}")
+
+        env_vars.update(
+            {
+                "SYCL_CACHE_PERSISTENT": "1",
+                "GMX_CUDA_GRAPH": "1",
+            }
+        )
+
+        # Run benchmark
+        if self.type == "rf":
+            command = [
+                str(self.gmx_path),
+                "mdrun",
+                "-s",
+                str(model_dir / "rf.tpr"),
+                "-nb",
+                "gpu",
+                "-update",
+                "gpu",
+                "-bonded",
+                "gpu",
+                "-ntmpi",
+                "1",
+                "-ntomp",
+                "1",
+                "-nobackup",
+                "-noconfout",
+                "-nstlist",
+                "100",
+                "-pin",
+                "on",
+            ]
+        else:  # type == "pme"
+            command = [
+                str(self.gmx_path),
+                "mdrun",
+                "-s",
+                str(model_dir / "pme.tpr"),
+                "-pme",
+                "gpu",
+                "-pmefft",
+                "gpu",
+                "-notunepme",
+                "-nb",
+                "gpu",
+                "-update",
+                "gpu",
+                "-bonded",
+                "gpu",
+                "-ntmpi",
+                "1",
+                "-ntomp",
+                "1",
+                "-nobackup",
+                "-noconfout",
+                "-nstlist",
+                "100",
+                "-pin",
+                "on",
+            ]
+
+        mdrun_result = run(
+            command,
+            add_sycl=True,
+        )
+        mdrun_result_output = mdrun_result.stderr.decode()
+        time = self._extract_execution_time(mdrun_result_output, type)
+
+        if options.verbose:
+            print(f"[{self.name()}-RF] Time: {time:.3f} seconds")
+
+        # Return results as a list of Result objects
+        return [
+            Result(
+                label=f"{self.name()}-{self.type}",
+                value=time,
+                unit="s",
+                passed=(mdrun_result.returncode == 0),
+                command=" ".join(map(str, command)),
+                env=env_vars,
+                stdout=self.conf_result.stderr.decode() + mdrun_result.stderr.decode(),
+                git_url=self.suite.git_url(),
+                git_hash=self.suite.git_tag(),
+            )
+        ]
+
+    def _extract_execution_time(self, log_content, benchmark_type):
```

Comment:
I wouldn't bother passing the benchmark_type just so that you can print error message differently. The framework already collects errors into a map indexed by the benchmark name.